Add RRM `Web_Tag`

nfmohit opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #8844

## Relevant technical choices

This PR adds the `Web_Tag` class for the Reader Revenue Manager module.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

techanvil commented on includes/Modules/Reader_Revenue_Manager/Tag_Matchers.php:
```diff
@@ -0,0 +1,38 @@
+	 */
+	public function regex_matchers() {
+		return array(
+			"/<script\\s+[^>]*src=['|\"]https?:\\/\\/news\\.google\\.com\\/swg\\/js\\/v1\\/swg-basic\\.js['|\"][^>]*>",
+			'/\\(self\\.SWG_BASIC=self\\.SWG_BASIC\\|\\|\\[\\]\\)\\.push/',
```
The first of these regexes is missing the closing forwardslash. Also, the backslashes don't need escaping:

```suggestion
			"/<script\s+[^>]*src=['|\"]https?:\/\/news\.google\.com\/swg\/js\/v1\/swg-basic\.js['|\"][^>]*>/",
			'/\(self\.SWG_BASIC=self\.SWG_BASIC\|\|\[\]\)\.push/',
```

nfmohit replied:
Updated, thank you!

techanvil commented on tests/phpunit/integration/Modules/Reader_Revenue_Manager/Web_TagTest.php:
```diff
@@ -0,0 +1,36 @@
+		$footer_html = $this->capture_action( 'wp_footer' );
+
+		$this->assertStringContainsString( 'Google Reader Revenue Manager snippet added by Site Kit', $footer_html );
+		$this->assertStringContainsString( 'script type="text/javascript" src="https://news.google.com/swg/js/v1/swg-basic.js" id="google_swgjs-js"></script>', $footer_html );
```
Super trivial, but it would be nice to include the opening bracket for this test, similar to other assertions elsewhere in the codebase.

```suggestion
		$this->assertStringContainsString( '<script type="text/javascript" src="https://news.google.com/swg/js/v1/swg-basic.js" id="google_swgjs-js"></script>', $footer_html ); // phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript
```

nfmohit replied:
This was an oversight, updated, thank you!

techanvil commented on tests/phpunit/integration/Modules/Reader_Revenue_ManagerTest.php:
```diff
@@ -186,11 +186,51 @@ public function test_on_deactivation() {
+		$footer_html = $this->capture_action( 'wp_footer' );
+
+		$this->assertStringContainsString( 'Google Reader Revenue Manager snippet added by Site Kit', $footer_html );
+		$this->assertStringContainsString( 'script type="text/javascript" src="https://news.google.com/swg/js/v1/swg-basic.js" id="google_swgjs-js"></script>', $footer_html );
```
As above:

```suggestion
		$this->assertStringContainsString( '<script type="text/javascript" src="https://news.google.com/swg/js/v1/swg-basic.js" id="google_swgjs-js"></script>', $footer_html ); // phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript
```

nfmohit replied:
Updated, thank you!

techanvil reviewed (changes requested):
Nice work here @nfmohit! I've left a few minor comments, I could just apply them directly but figured I'd send it back over to you for the sake of visibility.

nfmohit commented:
> Nice work here @nfmohit! I've left a few minor comments, I could just apply them directly but figured I'd send it back over to you for the sake of visibility.

I've addressed all the comments, thank you for the kind review, @techanvil !

techanvil reviewed (approved):
Thanks @nfmohit. LGTM!
